pkg/id/id.go

```diff
@@ -90,6 +100,14 @@ func NewAllocator(params *AllocatorParams) Allocator {
 	if allocator.step == 0 {
 		allocator.step = defaultAllocStep
 	}
+	var effectiveEnd uint64
+	effectiveEnd = math.MaxUint64
```

Comment:
uint32?

Reply:
Its type should be the same with `base` and `end`.

Reply:
What I mean is do we need to use `MaxKeyspaceID` instead of `math.MaxUint64`

Reply:
It will be fixed in another PR to be convenient to cherry-pick. https://github.com/tikv/pd/pull/9423